src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -53,9 +51,59 @@ class HttpGoogleCloudStorageDAO(
 
   val system = akka.actor.ActorSystem("system")
 
-  override def createBucket(userInfo: UserInfo, projectId: String, workspaceId: String): Unit = {
-    val inserter = getStorage(getBucketCredential(userInfo)).buckets.insert(projectId, new Bucket().setName(getBucketName(workspaceId)))
-    retry(inserter.execute,when500)
+  override def createBucket(userInfo: UserInfo, projectId: String, workspaceId: String, workspaceName: WorkspaceName): Unit = {
+    val bucketName = getBucketName(workspaceId)
+    val groups = getGroupDirectory.groups
+    insertGroup(groups,newGroup(bucketName,workspaceName,WorkspaceAccessLevel.Read)) {
```

Comment:
while this works, I do not like all the repetition and the rational behind it (rolling back) is hidden. It seems like you could map the list of access levels where you create the group for each and result in a map/tuple from entity to access level. Each element could also be a Try and you can test all for success then if any fail remove the successful ones.

I would like @abaumann or @jmthibault79 weigh in with opinions too. I don't know if it is worth that much of a fuss.